test/functional/testnvim.lua

```diff
@@ -24,6 +24,7 @@ M.nvim_set = (
   'set shortmess+=IS background=light noswapfile noautoindent startofline'
   .. ' laststatus=1 undodir=. directory=. viewdir=. backupdir=.'
   .. ' belloff= wildoptions-=pum joinspaces noshowcmd noruler nomore redrawdebug=invalid'
+  .. [[ statusline=%<%f\ %{%nvim_eval_statusline('%h%w%m%r',\ {'maxwidth':\ 30}).width\ >\ 0\ ?\ '%h%w%m%r\ '\ :\ ''%}%=%{%\ &showcmdloc\ ==\ 'statusline'\ ?\ '%-10.S\ '\ :\ ''\ %}%{%\ exists('b:keymap_name')\ ?\ '<'..b:keymap_name..'>\ '\ :\ ''\ %}%{%\ &ruler\ ?\ (\ &rulerformat\ ==\ ''\ ?\ '%-14.(%l,%c%V%)\ %P'\ :\ &rulerformat\ )\ :\ ''\ %}]]
```

Comment:
What is the intention of this expression?

Reply:
This is to fix a baseline statusline for the UI tests. Otherwise, every time we update the statusline option lots of unrelated UI tests will break. Instead, the UI tests use this expression for statusline (which was the closest estimate of the default statusline at the time of the refactor), and tests in statusline_spec assert the actual default statusline. So any unintentional change still gets captured, and reduces test churn when we want to make intentional changes. We had discussed it somewhere in this thread, I believe.